### PR TITLE
Feature/Join Idea Constraints

### PR DESCRIPTION
**Pull request by ac524**
This PR adds new features to **detect if a user has approval for an idea within a space** and adds constraints to views and api routes to **prevent a user from joining a new idea, if they are already an approved member of an exist idea**.

### Notes for Reviewer
- Main query logic to fetch data used in validation is built in to static model methods. See `Idea.getStatus` and `Interest.findUserApprovalInSpace`

### Reviewer Check List
**If the logged in user owns an idea or has been approved to collaborate within an `idea` for a `space`**
- [x] Logged in user will not see "Add Idea" on the space page
- [x] `GET /space/:space_id/ideas` redirect logged in user will be redirected back to the space page
- [x] `POST /space/:space_id/idea/:idea_id` Should error for logged in user
- [x] Logged in user should not see "join features" within any ideas of the space.
- [x] `POST /:space_id/idea/:idea_id/interest` Should error for logged in user
- [x] `PUT /:space_id/idea/:idea_id/interest` Should error for logged in user

**If the owner of an idea**
- [x] Should not see records of users approved for other ideas within management tools.
- [x] `PUT /:space_id/idea/:idea_id/interest/:user_id` with `{ "status": "approved" } Should error for logged in owner of idea.

### CODE REVIEW

**Comment by ac524** [comment]
I'll get up better seed data for testing tonight

File: views/idea.handlebars [hunk 1]
```diff
@@ -48,8 +54,8 @@
         {{/if}}
     </div>
 </div>
-
-{{#unless (is_prop_in idea.interests_status user_id)}}
+{{!-- {{#if can_join}} --}}
```
Comment (jmichaelbrown8) [inline views/idea.handlebars:57]: Do we want to uncomment this part so that it hides the lower "Become a collaborator" section here too?

**Review by jmichaelbrown8 — APPROVED** [review]
Looks good! Just had the one question about the commented out if statement in the idea.handlebars.

File: views/idea.handlebars (hunk 1 above)
Comment (ac524) [reply views/idea.handlebars]: Thanks for the catch!

**Pull request merged**